docs/tutorials/hackathon-prep-course/2-static-website.mdx

```diff
@@ -0,0 +1,100 @@
+
+To compile frontend assets like CSS and JavaScript into a Wasm module that can be deployed to ICP as a canister, `dfx` uses Rust boilerplate code to create an **asset canister**.
+
+:::info
+The terms 'asset canister' and 'frontend canister' are sometimes used interchangeably. **Asset** canister is typically used to describe the Rust code `dfx` uses in the background to compile a project's frontend assets into Wasm. **Frontend** canister is typically used as a general term to describe a project's frontend.
+:::
+
+Applications deployed on ICP can have frontends that:
+
+- Are deployed on ICP and communicate with backend canister(s) to create a fully onchain dapp.
+
+- Are hosted externally (off-chain) and communicate with backend canister(s) deployed on ICP.
+
+- Only provide web assets and do not communicate with a backend canister.
+
+In the previous tutorial, you deployed a project with a single backend canister and then interacted with that canister through the Candid API interface. In this tutorial, let's explore how a canister can host a static website that does not interact with a backend canister and is used only to serve web assets.
+
+## Crypto blog example
+
+[Open the ICP Ninja 'My Crypto Blog' project](https://icp.ninja/s/GQVBz). [View the ICP Ninja demo video](https://www.youtube.com/watch?v=19d8TrhWWQk) for more details about using the platform.
+
+Let's review the project's file structure:
+
+```bash
+├── frontend     # Folder containing the asset files for your dapp's frontend.
+│   ├──  public
+|   │   ├── favicon.ico
+│   ├──  src
+|   │   ├── main.jsx # Source code for a simple React application.
+│   ├──  index.css
+│   ├──  index.html
+│   ├──  package.json
+│   ├──  postcss.config.js
+│   ├──  tailwind.config.js
+│   ├──  vite.config.js
+├── dfx.json     # The configuration file for your Internet Computer dapp.
+├── package-lock.json
+├── package.json
+├── README.md    # Information about the project and using ICP Ninja.
+```
+
+### dfx.json​
+
+The `dfx.json` file for this project will closely resemble the one used for the [Hello, world!](./1-hello-world.mdx) example; however there is one key difference. Rather than defining what the canister is `"type": "motoko"`, this project will define the canister as `"type": "asset"` which prompts `dfx` to compile the canister's assets into a Wasm module using Rust boilerplate code.
+
+```json title=dfx.json file=../../references/samples/hosting/my_crypto_blog/dfx.json
+```
+
+### `frontend/src/main.jsx`
+
+This project's `main.jsx` file defines the source code for a simple app using React, Vite, and Tailwind CSS that the canister will host. This app will create a very basic blog website with three randomly generated placeholder blog post entries.
+
+```js reference file=../../references/samples/hosting/my_crypto_blog/frontend/src/main.jsx
```

Comment:
this rendering currently looks strange in the preview, see https://hhcgk-raaaa-aaaam-abahq-cai.icp0.io/docs/tutorials/hackathon-prep-course/static-website#frontendsrcmainjsx